cmd/vulcan-seekret/main.go

```diff
@@ -117,16 +117,23 @@ func main() {
 			return checkstate.ErrAssetUnreachable
 		}
 
-		repoPath := filepath.Join("/tmp", filepath.Base(targetURL.Path))
+		repoPath := filepath.Join(os.TempDir(), "repo")
 		if err := os.Mkdir(repoPath, 0755); err != nil {
 			return err
 		}
 
+		co := git.CloneOptions{
+			URL:   target,
+			Auth:  auth,
+			Depth: opt.Depth,
+		}
+		if opt.Branch != "" {
+			co.ReferenceName = plumbing.ReferenceName(path.Join("refs/heads", opt.Branch))
+		}
 		_, err = git.PlainClone(repoPath, false, &git.CloneOptions{
```

Comment:
I think you intended to use here the CloneOptions created above in the line: 125, so I guees you should modify theese lines by:
```go
_, err = git.PlainClone(repoPath, &co)
```

Reply:
Done, thanks!